recon/src/recon/tests.rs

```diff
@@ -1427,6 +1479,80 @@ async fn two_in_sync() {
     .await
 }
 
+#[test(tokio::test)]
+async fn dog_missing_sync() {
+    recon_test(expect![[r#"
+        cat: [a: A, b: B, c: C]
+        dog: [a: ∅, b: ∅, c: ∅]
+        -> interest_req((𝚨, 𝛀 ))
+            cat: [a: A, b: B, c: C]
+        <- interest_resp((𝚨, 𝛀 ))
+            dog: [a: ∅, b: ∅, c: ∅]
+        -> range_req({𝚨 h(a, b, c)#3 𝛀 })
+            cat: [a: A, b: B, c: C]
+        <- range_resp({𝚨 h(a, b, c)#3 𝛀 })
+            dog: [a: ∅, b: ∅, c: ∅]
+        <- value_req(a)
+            dog: [a: ∅, b: ∅, c: ∅]
+        -> listen_only
+            cat: [a: A, b: B, c: C]
+        <- value_req(b)
+            dog: [a: ∅, b: ∅, c: ∅]
+        -> value_resp(a: A)
+            cat: [a: A, b: B, c: C]
+        <- value_req(c)
+            dog: [a: A, b: ∅, c: ∅]
+        -> value_resp(b: B)
+            cat: [a: A, b: B, c: C]
+        <- listen_only
+            dog: [a: A, b: B, c: ∅]
+        -> value_resp(c: C)
+            cat: [a: A, b: B, c: C]
+        -> finished
+            cat: [a: A, b: B, c: C]
+        cat: [a: A, b: B, c: C]
+        dog: [a: A, b: B, c: C]
+    "#]])
+    .await;
+}
+
+#[test(tokio::test)]
+async fn cat_missing_sync() {
+    recon_test(expect![[r#"
+        cat: [a: ∅, b: ∅, c: ∅]
+        dog: [a: A, b: B, c: C]
+        -> interest_req((𝚨, 𝛀 ))
+            cat: [a: ∅, b: ∅, c: ∅]
+        <- interest_resp((𝚨, 𝛀 ))
+            dog: [a: A, b: B, c: C]
+        -> range_req({𝚨 h(a, b, c)#3 𝛀 })
+            cat: [a: ∅, b: ∅, c: ∅]
+        <- range_resp({𝚨 h(a, b, c)#3 𝛀 })
+            dog: [a: A, b: B, c: C]
+        -> value_req(a)
+            cat: [a: ∅, b: ∅, c: ∅]
+        -> value_req(b)
+            cat: [a: ∅, b: ∅, c: ∅]
+        <- value_resp(a: A)
```

Comment:
Along those lines.. in #243, I'm seeing some ordering changes to req/resp values cause failures and was hoping to get some help understanding what should happen and if it's a problem.

Reply:
I regenerated the expect tests for my PR and they seem to be consistent in the that order for now. Will cause some more conflicts for you merging back in.. hopefully it's not bad. It should be merged shortly.

Reply:
So its not perfectly deterministic but it does seem to be stable for a given implementation. Meaning I do not expect the tests to be flaky but changes in the performance of the code might change the order.

The test driver explicitly uses a single thread to drive the message exchange, but its still possible because of the async nature of the code that messages can be delivered in a different order. 

If we discover that the tests are somehow flaky we can explore ways to improve them. For example we could remove the assertions about state from each message and then allow for messages to be out of order using a new syntax. However I'd prefer to avoid that complexity if possible.

TL;DR we might have to update the expectations when we change to code but the tests should be stable.

Reply:
Sounds great, thanks.